### PR TITLE
SG-35165 Retry on URLError

### PR DESCRIPTION
**Pull request by carlos-villavicencio-adsk**
- Add retry only when encountering an URLError
- Adopt the existing strategy
- Add tests

### CODE REVIEW

File: shotgun_api3/shotgun.py [hunk 1]
```diff
@@ -4250,19 +4252,28 @@ def _send_form(self, url, params):
...
+        attempt = 1
+        while attempt <= self.MAX_ATTEMPTS:
+            # Perform the request
+            try:
+                resp = opener.open(url, params)
+                result = resp.read()
+                # response headers are in str(resp.info()).splitlines()
+            except urllib.error.URLError as e:
+                LOG.debug("Got a %s response. Waiting and retrying..." % e)
+                time.sleep(float(attempt) * self.BACKOFF)
```
Comment (julien-lang) [inline shotgun_api3/shotgun.py:4264]: The problem with this loop is that, in the worst case scenario, where every attempt failed, you end up sleeping before leaving the loop.

File: tests/test_client.py [hunk 2]
```diff
@@ -449,7 +449,7 @@ def test_call_rpc(self):
         self._mock_http(d, status=(504, "gateway timeout"))
         self.assertRaises(api.ProtocolError, self.sg._call_rpc, "list", a)
         self.assertEqual(
-            4,
+            3,
```
Comment (julien-lang) [inline tests/test_client.py:452]: what's this change?

Comment (carlos-villavicencio-adsk) [reply tests/test_client.py:452]: I standardize the number of retries to be reused on the library.

Comment (julien-lang) [reply tests/test_client.py:452]: Can you use `self.sgMAX_ATTEMPTS` instead or does it make sense to keep these 3 everywhere?

Comment (carlos-villavicencio-adsk) [reply tests/test_client.py:452]: I think it's feasible.

**Pull request merged**